### ISSUE TITLE
Blank content when using FancyZones

### ISSUE THREAD

**Issue by CrPO4** [flutter/flutter#159630]
### Steps to reproduce

1. Install Microsoft's FancyZones, setup any kind of zone,
2. Run a Flutter-based app (in my case it's Rustdesk but demo is done with Flutter Simple Demo, timecode: 0:12)
3. Bind app to a zone you created (timecode: 0:15)
4. Restart the app
5. Graphics get corrupted, no content is showing (0:22)
6. Move the app out of the zone and restart the app (timecode: 0:28)
7. Everything is fine again.

Reference video: https://www.youtube.com/watch?v=hr3hcWKMIhY

### Expected results

Everything shows up correctly, no blanking.

### Actual results

Content is blank, application is unusable.

### Code sample

There's no way for me to provide this piece, I'm not a developer of neither of the apps.

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>
(https://www.youtube.com/watch?v=hr3hcWKMIhY)](https://www.youtube.com/watch?v=hr3hcWKMIhY)
</details>

### Flutter Doctor output

There's no way for me to provide this piece, I'm not a developer of neither of the apps.

**Comment by darshankawar** [comment]
@CrPO4 
I am not well versed with Fancyzones utility you are using. It seems to be a window manager utility per https://learn.microsoft.com/en-us/windows/powertoys/fancyzones

Does it mean you are trying to have run flutter app in multi view windows ? If so, please check https://github.com/flutter/flutter/issues/142845 for reference to see if it is similar to your case or not.

**Comment by CrPO4** [comment]
@darshankawar 
You're right, FancyZones is somewhat related to what you'd call a window manager but "window organizer" would be is a bit closer to what it does.
Basically, it's an improved version of a built-in window snapping feature that is already present in Windows but it allows you to define zones instead of using default ones.

Now to the matter itself - no, I'm not trying to use multiwindows here, in every use case I only have one Flutter app launched and only one instance of it. All I do on demo is "snap" an existing window to the zone - basically, fixing it's location and size until "unsnapped".
If a Flutter app is being launched, and FancyZones tries to snap it immediately on launch (because it's been snapped before, FancyZones remembers that), then graphics get "blanked" as shown.
If you "snap" app's window and do not relaunch the app - it'd work flawlessly until relaunch.

**Comment by darshankawar** [comment]
Thanks for the update. I am unable to verify or provide more guidance on this, so based on the report, I will keep the issue open and label for team's tracking.

**Comment by JosephShall** [comment]
I found a workaround, I added my flutter app's exe name into excluded apps list of fancyzone like below.

![Image](https://github.com/user-attachments/assets/7de6dd53-a86b-4c4d-82ad-8d3763ffd6d6)

**Comment by CrPO4** [comment]
This is not a "true" workaround as it disables FancyZones completely for RustDesk.
It was also already found multiple times by users in [RustDesk issue.](https://github.com/rustdesk/rustdesk/issues/6756)

**Comment by JosephShall** [comment]
But It works for me

**Comment by mattbosak** [comment]
**Additional evidence -- render surface size appears to be locked at controller construction**

I've been chasing this in my own Flutter Windows app and wanted to add reproducible evidence that narrows down where the bug lives, since the original report couldn't include developer-level detail.

**Setup:** Standard Flutter Windows runner (unmodified `Win32Window` / `FlutterWindow` skeleton from `flutter create`). FancyZones with a zone bound to the app. Repro is 100% reliable on relaunch into the bound zone.

**What I tried at the embedder layer (all failed -- content remains blank):**

Reactive approaches in `FlutterWindow::MessageHandler`:
- `flutter_controller_->ForceRedraw()` on `WM_SIZE`, both before and after delegating to `Win32Window::MessageHandler`
- Same on `WM_WINDOWPOSCHANGED`, with an explicit `MoveWindow` on the child HWND
- `WM_TIMER` 100ms after first frame, calling `MoveWindow(child, …, h-1, …)` then `MoveWindow(child, …, h, …)` then `ForceRedraw()`
- `PostMessage(child_hwnd, WM_SIZE, SIZE_RESTORED, MAKELPARAM(w, h))` directly to the Flutter child HWND
- `SetWindowPos(child, …, SWP_FRAMECHANGED | SWP_NOZORDER | SWP_NOACTIVATE)`
- Posting a custom `WM_APP+1` from `SetNextFrameCallback` to drive a deferred resize cycle

Proactive approaches (initialize the engine at the correct size from the start):
- `Show()` early, pump messages with `PeekMessage` for 100/150/200ms, read `GetClientArea()`, then construct `FlutterViewController` with the post-snap size
- Same, with `SetNextFrameCallback` removed entirely
- Same, with `SetNextFrameCallback` retained as a no-op

Last-resort approach:
- Tear down `flutter_controller_` and rebuild it at the post-snap size on the first `WM_SIZE` after `OnCreate` (full `unique_ptr::reset()` + new `FlutterViewController`). This also failed.

**What the logs show:**

| Checkpoint | Parent HWND | Child HWND |
|---|---|---|
| Before `FlutterViewController` construction (post-pump) | 2559×1398 ✓ | -- |
| Immediately after `SetChildContent` | 2559×1398 ✓ | 2559×1398 ✓ |
| After FancyZones-induced `WM_SIZE` | 2559×1398 ✓ | 2559×1398 ✓ |

`WM_SIZE` fires twice during startup -- once at 1264×681 with `child_content_ == nullptr`, once at 2559×1398 also with `child_content_ == nullptr` (both before `OnCreate` completes). By the time the controller is built, the parent HWND is already at the correct 2559×1398.

**Despite all of the above, Flutter renders at 1264×681** - the original pre-snap size - and the rest of the client area is blank. The HWND geometry, child window geometry, and constructor arguments are all correct and consistent at 2559×1398.

**What this suggests:**

The render surface size appears to be committed somewhere below the HWND layer (swap chain / D3D render target) at a point that is not influenced by:
- the dimensions passed to `FlutterViewController(width, height, project)`,
- the size of the child HWND at construction time,
- any subsequent `WM_SIZE` delivered to the child HWND,
- `ForceRedraw()`,
- destroying and reconstructing the entire `FlutterViewController`.

The last point is the most surprising one and I think the most useful signal: a fresh controller, constructed *after* the window has settled at 2559×1398, with that exact size passed to the constructor, still renders at 1264×681. That points away from "missed resize message" and toward something cached at engine/process level -- possibly DPI resolved from the monitor at first-window-creation time, or render target dimensions captured before the controller's constructor args are consulted.

Can provide a minimal repro repo, ETW traces, or test specific patches if useful. This is currently a hard blocker for shipping to any user who has FancyZones configured, which is a non-trivial population.

Environment: Flutter 3.41.6 (stable), Windows 10 Build 19045.7184, PowerToys 0.98.1.
